appinventor/appengine/src/com/google/appinventor/client/actions/ShowShortcutsAction.java

```diff
@@ -29,32 +27,42 @@ public ShowShortcutsAction() {
         db.setGlassEnabled(true);
         db.setAnimationEnabled(true);
 
-        shortcutKeyHandler();
+        shortcutKeyHandler(this);
     }
 
     @Override
     public void execute() {
         VerticalPanel DialogBoxContents = new VerticalPanel();
         HTML message = new HTML(MESSAGES.KeyBoardShortcuts());
+        Button button = new Button("OK");
+        button.addClickHandler(event -> db.hide());
         DialogBoxContents.add(message);
+        DialogBoxContents.add(button);
         db.setWidget(DialogBoxContents);
         db.center();
         db.show();
     }
 
-    private void shortcutKeyHandler() {
-      Event.addNativePreviewHandler(new Event.NativePreviewHandler() {
-        @Override
-        public void onPreviewNativeEvent(Event.NativePreviewEvent event) {
-          NativeEvent nativeEvent = event.getNativeEvent();
-          if (event.getTypeInt() == Event.ONKEYDOWN && nativeEvent.getAltKey() && nativeEvent.getKeyCode() == 191 && !db.isShowing()) {
-            nativeEvent.preventDefault();
-            execute();
-          } else if (event.getTypeInt() == Event.ONKEYDOWN && db.isShowing()) {
-            nativeEvent.preventDefault();
-            db.hide();
+    private native void shortcutKeyHandler(ShowShortcutsAction action) /*-{
+      $wnd.document.addEventListener("keydown", function (event) {
+          if (event.altKey && event.key === "?") {
```

Comment:
This change seems to have broken Alt-? shortcut for Macs. I'm pretty sure it worked before. I may turn up more information in further testing.

Reply:
More insight into the Alt - ? on Mac: It looks like if I use either Alt-/ or Alt-shift-/ (i.e. ?), the / is the only key that is being taken seriously. My focus is moving to the component search bar rather than opening the help popup.

Is there any chance that there are circumstances where / is being checked before this check for Alt-/ or Alt-shift-/

(I will avoid adding a question mark to my sentence because this comment is confusing enough.)